File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> answer;
        
        int i, greater;
        
        for(auto& num1 : nums1) {
            i = 0;
            greater = -1;
            while(nums2[i++] != num1);
            for(;i < nums2.size(); i++) {
                if(nums2[i] > num1) {
                    greater = nums2[i];
                    break;
                }
            }
            answer.push_back(greater);
        }
        
        return answer;
    }
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        return [[y for y in nums2[nums2.index(x):] if y > x][0] if max(nums2[nums2.index(x):]) > x else -1  for x in nums1]